### PR TITLE
Added auxiliary heater climate entity

### PR DESCRIPTION
**Pull request by prvakt**
Auxiliary heater added as separate climate entity.
When started, and AC already running, AC must be switched off first. The same is valid for AC - when started and Auxiliary heater already running, auxiliary heater must be switched off first. If not, car will not change heater source

### CODE REVIEW

File: custom_components/myskoda/climate.py [hunk 1]
```diff
@@ -123,10 +130,116 @@ async def async_set_temperature(self, **kwargs):  # noqa: D102
...
             CapabilityId.AIR_CONDITIONING_SAVE_AND_ACTIVATE,
         ]
+
+
+class AuxiliaryHeater(MySkodaEntity, ClimateEntity):
+    """Auxiliary heater control for MySkoda vehicles."""
+
+    entity_description = ClimateEntityDescription(
+        key="auxiliary",
+        translation_key="auxiliary",
```
Comment (WebSpider) [inline custom_components/myskoda/climate.py:147]: Please rename this to `auxiliary_heater` to be more exact as to what it is about.

**Review by WebSpider — CHANGES_REQUESTED** [review]
Thanks for your contribution! :partying_face: 

Can you see if you can solve the dependencies in the PR description for the user? 
For instance by making the aux heater unavailable if the ac is running, etc?

The power of HA is in doing the hard stuff right by default, we should try to keep up that approach :)

**Comment by prvakt** [comment]
> Thanks for your contribution! 🥳
> 
> Can you see if you can solve the dependencies in the PR description for the user? For instance by making the aux heater unavailable if the ac is running, etc?
> 
> The power of HA is in doing the hard stuff right by default, we should try to keep up that approach :)

this is already solved - when user starts the Auxiliary heating and AC is running, AC is stopped.. same is valid for the opposite scenario -> I think this is better than making it unavailable

**Comment by WebSpider** [comment]
> > Thanks for your contribution! 🥳
> > Can you see if you can solve the dependencies in the PR description for the user? For instance by making the aux heater unavailable if the ac is running, etc?
> > The power of HA is in doing the hard stuff right by default, we should try to keep up that approach :)
> 
> sorry I guess I didn't write it clearly - this is already solved - when user starts the Auxiliary heating and AC is running, AC is stopped.. same is valid for the opposite scenario -> I think this is better than making it unavailable

Indeed, I missed that in the initial review, thanks for already doing that!

If we can be a bit more verbose to the user, so they can see what's going on, that would be even awesomer! :clinking_glasses:

File: custom_components/myskoda/translations/en.json [hunk 2]
```diff
@@ -93,6 +93,9 @@
         "climate": {
             "climate": {
                 "name": "Air Conditioning"
+            },
+            "auxiliary": {
```
[the remark this answers is not in the document]
Comment (prvakt) [reply custom_components/myskoda/translations/en.json]: yeah I just forgot to push this file in my previous commit

File: custom_components/myskoda/climate.py [hunk 3]
```diff
@@ -123,10 +130,117 @@ async def async_set_temperature(self, **kwargs):  # noqa: D102
...
+    async def async_set_temperature(self, **kwargs):  # noqa: D102
+        temp = kwargs[ATTR_TEMPERATURE]
+        await self.coordinator.myskoda.set_target_temperature(
+            self.vehicle.info.vin, temp
+        )
+        _LOGGER.info("Target temperature for auxiliary heater set.")
+
+    def required_capabilities(self) -> list[CapabilityId]:
+        return [
+            CapabilityId.AIR_CONDITIONING_HEATING_SOURCE_AUXILIARY,
```
[the remark this answers is not in the document]
Comment (prvakt) [reply custom_components/myskoda/climate.py:270]: Electric source is used for air-conditioning entity, and auxiliary source for auxiliary entity ... If I start auxiliary-heater with electric type instead, it will act exactly same as when air condition is turned on so I don't see any advantage of putting this capability inside.. 

Probably we will need to fine-tune the best capability, because from fixtures it seems like only Superb iV has AIR_CONDITIONING_HEATING_SOURCE_AUXILIARY. Kodiaq and Octavia is using  AUXILIARY_HEATING but that one is not available in Superb iV..

**Review by WebSpider — CHANGES_REQUESTED** [review]
Can you please remove the privacy sensitive information (vin) in the INFO logger statements? They are fine in DEBUG, but INFO is viewable by default.

**Review by WebSpider — APPROVED** [review]
Looks great, lets ship it and see if we need to finetune.
Thanks for another awesome contribution! :medal_sports:

**Comment by Tomanic-glitch** [comment]
I have updated to v1.10.0 but don't see any new entity for the Auxiliary heater even if my Skoda Kodiaq is equipped with it and it's present in the Myskoda app as well as in the old Skoda Connect integration

myskoda
skoda-connect
version1 10 0

**Comment by WebSpider** [comment]
Thanks for your report, ive created a new issue to address this
